File: iosApp/iosApp/state/DetailState.swift
//
//  DetailState.swift
//  iosApp
//
//  Created by Roberto García Romero on 5/2/24.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import shared

struct DetailStateSwift {
    var isLoading: Bool = true
    var logout: Bool = false
    var tapa: TapaItemBo? = nil
    var location: TapaLocation? = nil
    var hasLocationPermission: Bool? = nil
    var isGPSActive: Bool = false
    var isInRadius: Bool? = nil
    var voteStatus: VoteStatus = VoteStatus.unknown
}

File: shared/src/commonMain/kotlin/com/rgr/rutappa/app/state/DetailState.kt
package com.rgr.rutappa.app.state

import com.rgr.rutappa.domain.model.TapaItemBo

data class DetailState (
    val isLoading: Boolean = true,
    val logout: Boolean = false,
    val error: Boolean = false,
    val tapa: TapaItemBo? = null,
    val location: TapaLocation? = null,
    val hasLocationPermission: Boolean? = null,
    val isGPSActive: Boolean = false,
    val isInRadius: Boolean? = null,
    val voteStatus: VoteStatus = VoteStatus.UNKNOWN
)

enum class VoteStatus {
    UNKNOWN,
    CAN_VOTE,
    VOTED
}

data class TapaLocation(
    val latitude: String,
    val longitude: String
)